UCLCSSA/API-Server
# Add WeChat registration

Add ability to register with WeChat:
- Request `wechatOpenId` and `wechatSessionKey` from WeChat's `auth.code2session` API (see https://developers.weixin.qq.com/miniprogram/dev/api-backend/open-api/login/auth.code2Session.html).
- Generate `uclcssaSessionKey` based on `wechatOpenId` and `wechatSessionKey`.
- Store `uclcssaSessionKey`. If a user session already exists, update existing session. Otherwise, create new user session.

### Testing

Need help on doing integration testing for this WeChat registration flow.

Currently, I can verify that it does work by:
- Signing up for WeChat developer account, obtaining a temporary `appId` and `appSecret`.
- Provide `{ appId, appSecret, code }` via `wx.login` API.
- Check database for new `UserSession` (or updated).
- Wait for `{ uclcssaSessionKey }` to be returned.
